dll/cpl/usrmgr/users.c

```diff
@@ -650,6 +651,50 @@ UpdateUserProperties(HWND hwndDlg)
     NetApiBufferFree(pUserInfo);
 }
 
+void OnToggleRequireLogon(HWND hwndDlg)
+{
+    BOOL bIsChecked;
+    WCHAR szAutoAdminLogonValue[2]; 
+    HKEY hKey;
+    LONG lResult;
+    TCHAR szErrorMsg[256];
+
+    bIsChecked = IsDlgButtonChecked(hwndDlg, IDC_USERS_STARTUP_REQUIRE);
+
+    wcscpy(szAutoAdminLogonValue, (bIsChecked == BST_CHECKED) ? L"0" : L"1");
+
+    lResult = RegCreateKeyExW(HKEY_LOCAL_MACHINE,
+                              L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+                              0,
+                              NULL,
+                              REG_OPTION_NON_VOLATILE,
+                              KEY_SET_VALUE,
+                              NULL,
+                              &hKey,
+                              NULL);
+
+    if (lResult != ERROR_SUCCESS)
+    {
+        wsprintf(szErrorMsg, L"Failed to open or create Winlogon registry key for writing. Error code: %ld", lResult);
```

Comment:
```suggestion
        // TODO: Localize
        wsprintf(szErrorMsg, L"Failed to open or create Winlogon registry key for writing. Error code: %ld", lResult);
```
But I don't know whether it's worth having an error popup showing up to the user, or just something displayed in the debugger...

Reply:
I think it will be better to `DPRINT` what's in this line and show more friendly error message to the user.

Reply:
You decide, but note that we don't popup MessageBoxes for each and every failure code that can happen everywhere... Only the very important ones may deserve a msgbox.